elyra/pipeline/airflow/component_parser_airflow.py

```diff
@@ -46,149 +39,316 @@ class AirflowComponentParser(ComponentParser):
         # justification here: https://github.com/elyra-ai/elyra/issues/1912#issuecomment-879424452
         # properties.extend(self.get_runtime_specific_properties())
 
-        class_definition = self.get_class_def_as_string(content)
-        for arg in content.get('args'):
-            # For each argument to the init function, build a new parameter and add to existing
-            if arg in ['self', '*args', '**kwargs']:
-                continue
+        init_arguments = self._get_init_arguments(init_function)
+        for arg_name, arg_attributes in init_arguments.items():
+            data_type_from_ast = arg_attributes.get('data_type')
+
+            description = self._parse_description_from_docstring(arg_name, docstring)
+            data_type_parsed = self._parse_data_type_from_docstring(arg_name, docstring)
 
-            # Component parameter to be required to function correctly
-            required = True
-            value = None
-            if '=' in arg:
-                required = False  # We can use the default value if nothing is provided
-                arg, value = arg.split('=', 1)[:2]
-                value = ast.literal_eval(value)
-                if value and "\n" in str(value):
-                    value = value.replace("\n", " ")
-
-            # Search for data type (':type [param]:') in class docstring
-            type_regex = re.compile(f":type {arg}:" + r"([\s\S]*?(?=:type|:param|\"\"\"|'''|\.\.|\n))")
-            match = type_regex.search(class_definition)
-            data_type = match.group(1).strip() if match else "string"
-
-            # Search for description (':param [param]:') in class docstring
-            param_regex = re.compile(f":param {arg}:" + r"([\s\S]*?(?=:type|:param|\"\"\"|'''|\.\.))")
-            match = param_regex.search(class_definition)
-            description = match.group(1).strip().replace("\"", "'") if match else ""
-
-            # Amend description to include type information
-            description = self._format_description(description=description, data_type=data_type)
-
-            data_type_info = self.determine_type_information(data_type)
+            # Amend description to include type information as parsed, if available.
+            # Otherwise, include the type information determined from the AST parse
+            description = self._format_description(
+                description=description,
+                data_type=(data_type_parsed or data_type_from_ast)
+            )
+
+            # Standardize data type information
+            data_type_info = self.determine_type_information(data_type_parsed or data_type_from_ast)
             if data_type_info.undetermined:
-                self.log.debug(f"Data type from parsed data ('{data_type}') could not be determined. "
+                self.log.debug(f"Data type from parsed data ('{data_type_parsed}') could not be determined. "
```

Comment:
Since `data_type_info` _could_ come from `data_type_from_ast`, this should probably be: 
```python
('{data_type_parsed} or {data_type_from_ast}')
```